openslides/mediafiles/static/js/mediafiles/projector.js

```diff
@@ -18,11 +18,36 @@ angular.module('OpenSlidesApp.mediafiles.projector', [
 
 .controller('SlideMediafileCtrl', [
     '$scope',
+    '$timeout',
     'Mediafile',
-    function ($scope, Mediafile) {
+    function ($scope, $timeout, Mediafile) {
         // load mediafile object
         Mediafile.bindOne($scope.element.id, $scope, 'mediafile');
 
+        $scope.showPdf = true;
+
+        // Watch for page changes in the projector element. Adjust the page
+        // in the canvas scope, so the viewer can change the size automatically.
+        $scope.$watch(function () {
+            return $scope.element.page;
+        }, function () {
+            var canvasScope = angular.element('#pdf-canvas').scope();
+            if (canvasScope) {
+                canvasScope.pageNum = $scope.element.page;
+            }
+        });
+
+        // Watch for scale changes. If the scale is changed, reload the pdf
+        // viewer by just disable and re-enable it.
+        $scope.$watch(function () {
+            return $scope.element.scale;
+        }, function () {
+            $scope.showPdf = false;
+            $timeout(function () {
```

Comment:
This looks a little bit hacky but if we don't have a better solution we can merge it ...